backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/carousel/BpkCarousel.kt

```diff
@@ -0,0 +1,41 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.carousel
+
+import androidx.compose.foundation.layout.BoxScope
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import net.skyscanner.backpack.compose.carousel.internal.BpkCarouselImpl
+
+@Composable
+fun BpkCarousel(
+  imageContent: @Composable (BoxScope.(Int) -> Unit),
+  count: Int,
+  modifier: Modifier = Modifier,
+  currentImage: Int = 0,
+  onImageChanged: ((Int) -> Unit)? = null,
```

Comment:
I assume we always want to react on changing the image as it's the only way to update the content. If so, let's make it non-nullable

Reply:
it should be the callback AFTER the pager is scrolled to a different page, not the callback to get the content from?
https://www.figma.com/file/yN0hFyZlKL0Jwbpi0rEKYT/Backpack-Beta?node-id=8335%3A38592&t=kaBaEYMePlbY8sWh-0